controllers/etcd_controller.go

```diff
@@ -1123,11 +1150,177 @@ func (r *EtcdReconciler) waitUntilStatefulSetReady(sts *appsv1.StatefulSet) (*ap
 		if err := health.CheckStatefulSet(ss); err != nil {
 			return gardenerretry.MinorError(err)
 		}
+		if ss.Status.Replicas != *ss.Spec.Replicas {
+			return gardenerretry.MinorError(fmt.Errorf("ss not scaled"))
+		}
 		return gardenerretry.Ok()
 	})
 	return ss, err
 }
 
+func (r *EtcdReconciler) prepareEtcdForScaleDown(etcd *druidv1alpha1.Etcd, ss *appsv1.StatefulSet, svc *v1.Service) (*corev1.PersistentVolumeClaimList, error) {
```

Comment:
Let's follow `ctx` as parameter for new function we add where we require context.
```suggestion
func (r *EtcdReconciler) prepareEtcdForScaleDown(ctx Context.context, etcd *druidv1alpha1.Etcd, ss *appsv1.StatefulSet, svc *v1.Service) (*corev1.PersistentVolumeClaimList, error) {
```